v2/apiserver/internal/core/jobs.go

```diff
@@ -522,8 +515,9 @@ func (j *jobsService) WatchStatus(
 				log.Printf("error retrieving event %q from store: %s", eventID, err)
 				return
 			}
+			job, _ := event.Worker.Job(jobName)
```

Comment:
We're not checking if the job exists. We know it does. We checked that well before we entered this loop.